var/spack/repos/builtin/packages/mrtrix3/package.py

```diff
@@ -27,11 +31,14 @@ class Mrtrix3(Package):
     depends_on("py-numpy", type=("build", "run"))
     depends_on("glu")
     depends_on("qt+opengl@4.7:")
-    depends_on("eigen")
+    depends_on("eigen@3.3", when="@3.0.3")
+    depends_on("eigen@3.4", when="@3.0.4:")
```

Comment:
Is it necessary to pin the dependencies this specifically? Or can the ranges be broader?

Reply:
According to their [docs](https://mrtrix.readthedocs.io/en/dev/installation/build_from_source.html) they require eigen >= 3.2 with >= 3.3 recommended. These docs are actually slightly out-of-date as 3.0.4 does in fact require 3.4 to build.

Reply:
Greater or equal is expressed as:
```suggestion
    depends_on("eigen@3.3:", when="@3.0.3")
    depends_on("eigen@3.4:", when="@3.0.4:")
```
(note the additional `:`).

Reply:
Ah, sorry I forgot to mention that 3.0.3 won't compile with with eigen@3.4 due to conflicting declarations (which isn't mentioned in the docs).

Reply:
In that case, can you add a comment about that and use a range for the other directive
```suggestion
    # <Brief comment explaining why only 3.3>
    depends_on("eigen@3.3", when="@3.0.3")
    depends_on("eigen@3.4:", when="@3.0.4:")
```
?

Reply:
Ok, should be resolved with latest commit.  I also put a version upper bound on the patch since it sounds like it should be fixed upstream shortly.